Refs #25217 - Refactor notifications prop-types

sharvit opened the pull request.

Commit:
Refs #25217 - Refactor notifications prop-types

* Adds snapshot testing for notifications toggle-icon

ohadlevy commented on webpack/assets/javascripts/react_app/components/notifications/toggleIcon/index.js:
```diff
@@ -1,7 +1,9 @@
+import { noop } from '../../../common/helpers';
 
-export default ({ hasUnreadMessages, onClick }) => {
+const ToggleIcon = ({ hasUnreadMessages, onClick }) => {
```
shouldn't this trigger a snapshot change? (dont we have any?)

sharvit commented:
@ohadlevy Seems like we don't have a snapshot testing for it.
We do have a full mount testing (without snapshots) for the notifications itself.

* I moved the `toggleIcon/index.js` to `ToggleIcon/ToggleIcon.js`
* Added a `ToggleIcon/ToggleIcon.test.js`

ohadlevy commented:
[test foreman]
